src/routes.js

```diff
@@ -41,21 +44,41 @@ router.post('/coffeegrams', async ctx => {
     shop: form['shop']
   });
 
-  ctx.request.flash('success', "Your new Coffeegram is up!");
+  ctx.request.flash('success', "Your new Coffeegram is up");
 
   ctx.redirect(`/users/${ctx.state.currentUser.username}`);
 });
 
 router.get('/coffeegrams/:id', async ctx => {
   var coffeegram = await Coffeegram.findById(ctx.params.id);
-  if(!coffeegram) {
+  if (!coffeegram) {
     throw new NotFound();
   }
-  await ctx.render('edit', {
+  await ctx.render('gram', {
     coffeegram
   });
 });
 
+router.post('/coffeegrams/:id', async ctx => {
+  var coffeegram = await Coffeegram.findById(ctx.params.id);
+  if (!coffeegram) {
+    throw new NotFound();
+  }
+  if (coffeegram.userId.toString() !== ctx.state.currentUser.id) {
```

Comment:
Ah, apparently this is the wrong way to compare ObjectIDs. Try the `.equals()` method on ObjectID:

```
if (!coffeegram.userId.equals(ctx.state.currentUser.id)) {
```